velox/exec/tests/GroupedExecutionTest.cpp

```diff
@@ -283,6 +284,150 @@ TEST_F(GroupedExecutionTest, groupedExecutionWithOutputBuffer) {
   EXPECT_EQ(18, taskStats.pipelineStats[1].operatorStats[1].inputVectors);
 }
 
+DEBUG_ONLY_TEST_F(
+    GroupedExecutionTest,
+    groupedExecutionWithHashJoinSpillCheck) {
+  // Create source file to read as split input.
+  auto vectors = makeVectors(24, 20);
+  auto filePath = TempFilePath::create();
+  writeToFile(filePath->path, vectors);
+
+  struct {
+    bool enableSpill;
+    bool mixedExecutionMode;
+    int expectedNumDrivers;
+    bool expectedSpill;
+
+    std::string debugString() const {
+      return fmt::format(
+          "enableSpill {}, mixedExecutionMode {}, expectedNumDrivers {}, expectedSpill {}",
+          enableSpill,
+          mixedExecutionMode,
+          expectedNumDrivers,
+          expectedSpill);
+    }
+  } testSettings[] = {
+      {false, false, 12, false},
+      {false, true, 9, false},
+      {true, false, 12, true},
+      {true, true, 9, false}};
+
+  for (const auto& testData : testSettings) {
+    SCOPED_TRACE(testData.debugString());
+
+    auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+    core::PlanNodeId probeScanNodeId;
+    core::PlanNodeId buildScanNodeId;
+
+    PlanBuilder planBuilder(planNodeIdGenerator, pool_.get());
+    planBuilder.tableScan(rowType_)
+        .capturePlanNodeId(probeScanNodeId)
+        .project({"c3 as x", "c2 as y", "c1 as z", "c0 as w", "c4", "c5"});
```

Comment:
Why do we need this project?

Reply:
This is used for join as both probe and build use the same set of columns.